src/c++/perf_analyzer/concurrency_worker.cc

```diff
@@ -160,7 +160,6 @@ ConcurrencyWorker::PrepAndSendInferRequests()
 void
 ConcurrencyWorker::RestoreFreeCtxId(uint32_t ctx_id)
 {
-  // FIXME TMA-1023 we are clearly pushing and not popping in some cases
```

Comment:
We now always pop on line 154 and push for sync on 156 and push for async here. Before for certain sequence cases we would push and not pop.